src/gui/rss/rsswidget.cpp

```diff
@@ -559,6 +559,41 @@ bool RSSWidget::eventFilter(QObject *obj, QEvent *event)
     return false;
 }
 
+void RSSWidget::convertRelativePathToAbsolute(QString &html, const QString &basePath) const
+{
+    QRegularExpression rx;
+    rx.setPatternOptions(QRegularExpression::CaseInsensitiveOption);
+    // href regex
+    rx.setPattern(
+        uR"(((<a\s+[^>]*?href|<img\s+[^>]*?src)\s*=\s*["'])((https?|ftp):)?(\/\/[^\/]*)?(\/?[^\/"].*?)(["']))"_s);
+
+    QString normalizedBasePath = basePath.endsWith(u'/') ? basePath : basePath + u'/';
+    QRegularExpressionMatchIterator iter = rx.globalMatch(html);
+
+    while (iter.hasNext())
+    {
+        QRegularExpressionMatch match = iter.next();
+        const QString &fullMatch = match.captured(0);
+        const Qstring &prefix = match.captured(1);
+        const QString &scheme = match.captured(4);
+        const QString &host = match.captured(5);
+        QString relativePath = match.captured(6);
+        const QString &suffix = match.captured(7);
+        if (relativePath.startsWith(u'/'))
+            relativePath = relativePath.mid(1);
+        if (!scheme.isEmpty() && !host.isEmpty()) // already absolute path
+            continue;
+        else if (!host.isEmpty())
+            normalizedBasePath = u"http:" + host;
+        else if (!scheme.isEmpty())
+            break; // invalid url, should never happen
```

Comment:
It is better to reorder the code to separate the special conditions and avoid unnecessary calculations:
```suggestion
        const QString &scheme = match.captured(4);
        const QString &host = match.captured(5);
        if (!scheme.isEmpty())
        {
            if (host.isEmpty())
                break; // invalid URL, should never happen
                
            // already absolute path
            continue;
        }

        QString relativePath = match.captured(6);
        if (relativePath.startsWith(u'/'))
            relativePath = relativePath.mid(1);

        if (!host.isEmpty())
            normalizedBasePath = u"http:" + host;

        const QString &fullMatch = match.captured(0);
        const QString &prefix = match.captured(1);
        const QString &suffix = match.captured(7);
```